packages/material-ui/src/SvgIcon/SvgIcon.js

```diff
@@ -1,61 +1,75 @@
 import capitalize from '../utils/capitalize';
+import useThemeProps from '../styles/useThemeProps';
+import experimentalStyled from '../styles/experimentalStyled';
+import { getSvgIconUtilityClass } from './svgIconClasses';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    userSelect: 'none',
-    width: '1em',
-    height: '1em',
-    display: 'inline-block',
-    fill: 'currentColor',
-    flexShrink: 0,
-    fontSize: theme.typography.pxToRem(24),
-    transition: theme.transitions.create('fill', {
-      duration: theme.transitions.duration.shorter,
-    }),
-  },
-  /* Styles applied to the root element if `color="primary"`. */
-  colorPrimary: {
-    color: theme.palette.primary.main,
-  },
-  /* Styles applied to the root element if `color="secondary"`. */
-  colorSecondary: {
-    color: theme.palette.secondary.main,
-  },
-  /* Styles applied to the root element if `color="action"`. */
-  colorAction: {
-    color: theme.palette.action.active,
-  },
-  /* Styles applied to the root element if `color="error"`. */
-  colorError: {
-    color: theme.palette.error.main,
-  },
-  /* Styles applied to the root element if `color="disabled"`. */
-  colorDisabled: {
-    color: theme.palette.action.disabled,
-  },
-  /* Styles applied to the root element if `fontSize="inherit"`. */
-  fontSizeInherit: {
-    fontSize: 'inherit',
-  },
-  /* Styles applied to the root element if `fontSize="small"`. */
-  fontSizeSmall: {
-    fontSize: theme.typography.pxToRem(20),
-  },
-  /* Styles applied to the root element if `fontSize="large"`. */
-  fontSizeLarge: {
-    fontSize: theme.typography.pxToRem(35),
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...(styleProps.color !== 'inherit' && styles[`color${capitalize(styleProps.color)}`]),
+    ...styles[`elevation${capitalize(styleProps.fontSize)}`],
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { color, fontSize, classes } = styleProps;
+
+  const slots = {
+    root: [
+      'root',
+      color !== 'inherit' && `color${capitalize(color)}`,
+      `fontSize${capitalize(fontSize)}`,
+    ],
+  };
+
+  return composeClasses(slots, getSvgIconUtilityClass, classes);
+};
+
+const SvgIconRoot = experimentalStyled(
+  'svg',
+  {},
+  {
+    name: 'MuiSvgIcon',
+    slot: 'Root',
+    overridesResolver,
   },
-});
+)(({ theme, styleProps }) => ({
+  /* Styles applied to the root element. */
+  userSelect: 'none',
+  width: '1em',
+  height: '1em',
+  display: 'inline-block',
+  fill: 'currentColor',
+  flexShrink: 0,
+  transition: theme.transitions.create('fill', {
+    duration: theme.transitions.duration.shorter,
+  }),
+  fontSize: {
+    inherit: 'inherit',
+    small: theme.typography.pxToRem(20),
+    medium: theme.typography.pxToRem(24),
+    large: theme.typography.pxToRem(35),
+  }[styleProps.fontSize],
+  color: {
```

Comment:
We are not able to support dynamic colors with this. I'd recommend refactoring to:
```
...(styleProps.color !== 'disabled' && styleProps.color !== 'inherit' && {
  color: theme.palette[styleProps.color].main,
}),
...(styleProps.color === 'disabled' && {
  color: theme.palette.action.disabled,
}),
```

Reply:
I was wondering about this. The objective was to retain the previous behavior. I wonder if we shouldn't remove the color prop.

Reply:
That makes sense too... I don't have strong opinion. My thinking so far was, if we can make the color work automatically for other scenarios, I'd do it

Reply:
It seems that the SvgIcon color prop has no business logic, it's a CSS helper with an inconsistency mapping between prop => design token. It also seems that SvgIcon isn't a CSS helper like Typography so shouldn't have the system props.

Reply:
Ok, then let's leave it as you had for now and deprecate the prop, so we can remove it in the v6 maybe in favor of the sx prop.